CONTRIBUTING.md

```diff
@@ -177,6 +177,23 @@ To maintain understandable code and avoid circular dependencies it is important
 - **templates:** Golang templates for generating the html output.
 - **vendor:** External code that Gitea depends on.
 
+## API v1
+
+The API is documented by [swagger](http://gitea.com/api/swagger) and is based on [GitHub API v3](https://developer.github.com/v3/).
+So if there exist a endpoint on GitHub API it should be the same for gitea. If GitHub dont provide a functionality Gitea does, a own endpoint can cover this. If a information is on Gitea but not on GitHub API a new field is used wich doesnt coldie with GitHub ones.
+
+All expected results (errors, success, fail messages) should be documented ([example](https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/issue.go#L319-L327)).
+
+All JSON input types must be defined as struct in `models/structs/` ([example](https://github.com/go-gitea/gitea/blob/master/modules/structs/issue.go#L76-L91)) and reverenced in [routers/api/v1/swagger/options.go](https://github.com/go-gitea/gitea/blob/master/routers/api/v1/swagger/options.go), they can be used then as follow: ([example](https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/issue.go#L318)).
+
+All JSON responces must be defined as struct in `models/structs/` ([example](https://github.com/go-gitea/gitea/blob/master/modules/structs/issue.go#L36-L68)) and reverenced in its category in `routers/api/v1/swagger/` ([example](https://github.com/go-gitea/gitea/blob/master/routers/api/v1/swagger/issue.go#L11-L16)), they can be used the as follow: ([example](https://github.com/go-gitea/gitea/blob/master/routers/api/v1/repo/issue.go#L277-L279))
+
+ * GET endpoints return status `OK (200)`,
+ * POST endpoints return status `Created (201)` and
+ * DELETE endpoints return status `No Content (204)`
+
```

Comment:
PUT and PATCH?
I think there are exceptions to this codes, at least we used others today (but that could be wrong).
Also it would be good with something on when to use PUT (Idempotent) vs POST (and PATCH).

Reply:
* `POST` Creat thins  -> some return the created object also .. but it differs :( **Created (201)**
* `PUT` add/assign existing obejcts(e.g. User/Lable) (user to org-team) -> return no contend **No Content (204)**
* `PATCH` edit/alter an existing object (e.g. update a hook)   -> return full changed object **OK (200)**

Reply:
of course some of your APIs differ -> O i forgot the backwards compatibelity ...